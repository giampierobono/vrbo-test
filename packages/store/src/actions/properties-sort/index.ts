export * from "./properties-sort.actions";
