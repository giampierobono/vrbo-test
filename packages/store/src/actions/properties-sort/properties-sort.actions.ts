import { ActionWithPayload, PropertiesSortByConfig } from "@vrbo/data-models";

export enum PropertiesSortActions {
  SetSortByConfig = "[PropertiesSort] Set new sort param"
}

export const setNewPropertiesSortAction = (
  sortBy: PropertiesSortByConfig
): ActionWithPayload<PropertiesSortByConfig> => ({
  type: PropertiesSortActions.SetSortByConfig,
  payload: sortBy
});
