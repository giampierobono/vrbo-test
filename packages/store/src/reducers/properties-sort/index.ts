export * from "./properties-sort.reducer";
