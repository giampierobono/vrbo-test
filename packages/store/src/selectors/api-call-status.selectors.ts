import { GlobalState } from "../global-state.model";

export const getPropertiesListLoadingStatus = (state: GlobalState): boolean =>
  state.apiCallStatus.propertiesPollLoading;
