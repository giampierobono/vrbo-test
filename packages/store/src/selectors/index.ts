export * from "./api-call-status.selectors";
export * from "./property.selectors";
