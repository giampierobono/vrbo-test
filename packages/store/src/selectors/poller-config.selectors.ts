import { PollerConfig } from "@vrbo/data-models";
import { GlobalState } from "../global-state.model";

export const getPollerConfig = (state: GlobalState): PollerConfig =>
  state.pollerConfig.config;
