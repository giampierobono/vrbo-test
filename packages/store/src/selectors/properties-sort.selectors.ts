import { PropertiesSortByConfig } from "@vrbo/data-models";
import { GlobalState } from "../global-state.model";

export const getPropertiesSortByConfig = (
  state: GlobalState
): PropertiesSortByConfig => state.propertiesListSortBy.sortConfig;
