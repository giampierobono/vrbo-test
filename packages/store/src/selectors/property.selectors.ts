import { PropertiesSortBy, Property } from "@vrbo/data-models";
import { createSelector } from "reselect";
import { GlobalState } from "../global-state.model";

const getPropertiesList = (
  state: GlobalState,
  props: { sortBy: PropertiesSortBy }
) => state.propertiesList.properties;

export const makeGetOrderedPropertiesList = () =>
  createSelector(
    [getPropertiesList],
    (
      propertiesList: Property[],
      props: { sortBy: PropertiesSortBy }
    ): Property[] => {
      switch (props.sortBy) {
        case PropertiesSortBy.Title: {
          return propertiesList.sort((a, b) =>
            a.title === b.title ? 0 : a.title > b.title ? 1 : 0
          );
        }
        default: {
          return propertiesList;
        }
      }
    }
  );
