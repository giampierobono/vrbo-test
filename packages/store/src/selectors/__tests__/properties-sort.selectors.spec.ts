import { mockState } from "../../../../../jest-global-mocks";
import { getPropertiesSortByConfig } from "../properties-sort.selectors";

describe("Properties sort config", () => {
  describe("getPropertiesSortByConfig", () => {
    it("should return correct value", () => {
      expect(getPropertiesSortByConfig(mockState)).toMatchSnapshot();
    });
  });
});
