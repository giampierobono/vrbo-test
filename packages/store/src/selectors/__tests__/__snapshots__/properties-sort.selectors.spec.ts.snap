// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties sort config getPropertiesSortByConfig should return correct value 1`] = `"Title"`;
